.github/workflows/release.yml

```diff
@@ -8,57 +8,45 @@ on:
     types:
       - completed
 
-env:
-  GOPROXY: https://proxy.golang.org/
+permissions:
+  contents: write
 
 jobs:
-  Release:
+  go-version:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.go-version.outputs.version }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: go-version
+        run: echo "::set-output name=version::$(cat ./.go-version)"
+  release-notes:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
         with:
           fetch-depth: 0
-      - name: Read go version
-        id: go-version
-        run: |
-          content=`cat ./.go-version`
-          echo "::set-output name=content::$content"
-      - uses: actions/setup-go@v2
-        with:
-          # TODO: Replace with go-version-from-file when it is supported
-          # https://github.com/actions/setup-go/pull/62
-          go-version: ${{ steps.go-version.outputs.content }}
-      - uses: hashicorp/setup-hc-releases@v1
-        with:
-          github-token: ${{ secrets.HC_RELEASES_TOKEN }}
-          version: 0.11.4
-      - uses: hashicorp/setup-signore@v2
-        with:
-          github-token: ${{ secrets.SETUP_SIGNORE_GITHUB_TOKEN }}
-          signer: interim_signing_subkey_768B676
-      - name: Release Notes
-        run: sed -n -e "1{/# /d;}" -e "2{/^$/d;}" -e "/# \[$(git describe --abbrev=0 --exclude="$(git describe --abbrev=0 --match='v*.*.*' --tags)" --match='v*.*.*' --tags | tr -d v)/q;p" CHANGELOG.md > /tmp/RELEASE-NOTES.md
-      - uses: aws-actions/configure-aws-credentials@v1
-        with:
-          aws-access-key-id: ${{ secrets.TF_PROVIDER_RELEASE_AWS_ACCESS_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.TF_PROVIDER_RELEASE_AWS_SECRET_ACCESS_KEY }}
-          aws-region: us-east-1
-          role-to-assume: ${{ secrets.TF_PROVIDER_RELEASE_AWS_ROLE_ARN }}
-          role-duration-seconds: 3600
-          # TODO: allow session tagging once IAM permission changes addressed
-          # Reference: https://github.com/hashicorp/hc-releases/issues/124
-          role-skip-session-tagging: true
-      - name: goreleaser release
-        uses: goreleaser/goreleaser-action@v2
+      - name: Generate Release Notes
+        run: sed -n -e "1{/# /d;}" -e "2{/^$/d;}" -e "/# $(git describe --abbrev=0 --exclude="$(git describe --abbrev=0 --match='v*.*.*' --tags)" --match='v*.*.*' --tags | tr -d v)/q;p" CHANGELOG.md > release-notes.txt
```

Comment:
previously i added the escaped open bracket in `\[$(git describe...` so that sed would catch the text between the unreleased release notes and the previous tag which uses the pattern `## [0.x.x]...` since without it it seems to cut all the release notes back to 0.1.0. i've seen the pattern you've provided in use before though so do you recommend a way to verify locally?

Reply:
You can locally create the "next release" tag, e.g. `git tag v0.9.0` then execute the `run` line. Just be sure to delete the local tag, otherwise weird things could happen. This GitHub Action job is setup to run on Linux, so any differences between the `sed` command on Linux and your local platform may complicate things, however I think this was written in a way that works fine between macOS and Linux just fine (at least for me).

Since you folks are using Markdown links in your CHANGELOG currently, the `[` would need to be included in the handling here, which seems to work as expected.

```console
$ git tag v0.9.0
$ sed -n -e "1{/# /d;}" -e "2{/^$/d;}" -e "/# \[$(git describe --abbrev=0 --exclude="$(git describe --abbrev=0 --match='v*.*.*' --tags)" --match='v*.*.*' --tags | tr -d v)/q;p" CHANGELOG.md
BUG FIXES:

* provider: Ensure `darwin/arm64` platform is included in releases
* Prevent `terraform plan` showing that a resource must be replaced immediately after creation ([#306](https://github.com/hashicorp/terraform-provider-awscc/issues/306))
* Prevent errors like `An unexpected error was encountered trying to read an attribute from the state.` during resource read ([#306](https://github.com/hashicorp/terraform-provider-awscc/issues/306))

$ 
```

```suggestion
        run: sed -n -e "1{/# /d;}" -e "2{/^$/d;}" -e "/# \[$(git describe --abbrev=0 --exclude="$(git describe --abbrev=0 --match='v*.*.*' --tags)" --match='v*.*.*' --tags | tr -d v)/q;p" CHANGELOG.md > release-notes.txt
```